packages/ods-react/src/components/tree-view/src/components/tree-view-node/TreeViewNode.tsx

```diff
@@ -0,0 +1,161 @@
+import { TreeView as VendorTreeView, useTreeViewContext } from '@ark-ui/react/tree-view';
+import classNames from 'classnames';
+import { type ComponentPropsWithRef, type JSX, type KeyboardEvent, type ReactNode, type Ref, type RefAttributes, forwardRef, useRef } from 'react';
+import { Checkbox, CheckboxControl } from '../../../../checkbox/src';
+import { ICON_NAME, Icon } from '../../../../icon/src';
+import style from './treeViewNode.module.scss';
+
+export interface TreeViewItem<TCustom = Record<string, never>> {
+  id: string;
+  name: string;
+  children?: Array<TreeViewItem<TCustom>>;
+  icon?: ReactNode;
+  customRendererData?: TCustom;
+}
+
+export type TreeViewCustomRendererArg<TCustom = Record<string, never>> = {
+  item: TreeViewItem<TCustom>;
+  name: string;
+  customData?: TCustom;
+};
+
+interface TreeViewNodeProp<TCustom = Record<string, never>> extends ComponentPropsWithRef<'div'> {
+  item: TreeViewItem<TCustom>;
+  indexPath?: number[];
+  customBranchRenderer?: (arg: TreeViewCustomRendererArg<TCustom>) => JSX.Element;
+  customItemRenderer?: (arg: TreeViewCustomRendererArg<TCustom>) => JSX.Element;
+  multiple?: boolean;
+}
+
+function TreeViewNodeInner<TCustom = Record<string, never>>({
+  className,
+  item,
+  indexPath = [],
+  customBranchRenderer,
+  customItemRenderer,
+  ...props
+}: TreeViewNodeProp<TCustom>, ref: Ref<HTMLDivElement>): JSX.Element {
+  const hasChildren = Array.isArray(item.children) && item.children.length > 0;
+  const { getTreeProps } = useTreeViewContext();
+  const ariaMultiselectable = getTreeProps()?.['aria-multiselectable'] as boolean | 'true' | 'false' | undefined;
+  const multiple = ariaMultiselectable === true || ariaMultiselectable === 'true';
+  const checkboxRef = useRef<HTMLLabelElement | null>(null);
+  const handleKeyDown = (event: KeyboardEvent<HTMLDivElement>): void => {
+    if (!multiple) {
+      return;
+    }
+    if (event.key === ' ' || event.key === 'Enter') {
+      event.preventDefault();
+      checkboxRef.current?.click();
+    }
+  };
+  const CheckboxIndicator = (
+    <VendorTreeView.NodeCheckboxIndicator
+      fallback={
+        <Checkbox checked={ false }>
+          <CheckboxControl />
+        </Checkbox>
+      }
+      indeterminate={
+        <Checkbox checked={ 'indeterminate' }>
+          <CheckboxControl />
+        </Checkbox>
+      }>
+      <Checkbox checked>
+        <CheckboxControl />
+      </Checkbox>
+    </VendorTreeView.NodeCheckboxIndicator>
+  );
+  return (
+    <VendorTreeView.NodeProvider node={ item } indexPath={ indexPath }>
+      { hasChildren ? (
+        <VendorTreeView.Branch
+          className={ classNames(style['tree-view-node'], className) }
+          data-ods="tree-view-branch"
```

Comment:
hum should be `tree-view-node`, though it might be interesting to differenciate both in some way